fabric-network/test/impl/query/queryhandlers.js

```diff
@@ -56,19 +66,151 @@ describe('QueryHandlers', () => {
 		sinon.restore();
 	});
 
-	describe('SingleQueryHandler', () => {
+	describe('SingleQueryHandler with one peer', () => {
+		let handler;
+
+		beforeEach(() => {
+			handler = new SingleQueryHandler([endorser1]);
+		});
+
+		it('returns peer valid results - all good', async () => {
+			query.send.resolves(validProposalResponse);
+
+			const result = await handler.evaluate(query);
+			expect(result).to.be.equal(success);
+		});
+
+		it('uses first peer', async () => {
+			query.send.resolves(validProposalResponse);
+
+			await handler.evaluate(query);
+			expect(handler._currentPeerIndex).to.be.equal(0);
```

Comment:
The tests should typically be ensuring externally visible behaviour. Asserting on internal state like this means that any changes to the internal implementation will break the tests, making the code harder to refactor and the tests less useful as a regression suite since you can't change the code without changing the tests. You should (with the exception of perhaps some changing to stubbing/mocking) be able to refactor the implementation and continue using the same tests to ensure you've not introduced regressions. Look at how the original unit tests did it, and I would suggest just going back to them (or something very similar to them) with some minor changes to the stubbing/mocking to make them run again.